packages/engine/src/assets/classes/Asset.ts

```diff
@@ -0,0 +1,42 @@
+import { AssetClass } from '../enum/AssetClass'
+
+export type AssetId = string
+
+export class Dependency {
+  assets: Asset[]
+  loadCondition?: () => boolean
+  unloadCondition?: () => boolean
+  loaded: boolean
+  constructor(_assets: Asset[], _loadCondition?: () => boolean, _unloadCondition?: () => boolean) {
+    this.assets = _assets
+    if (_loadCondition) this.loadCondition = _loadCondition
+    if (_unloadCondition) this.unloadCondition = _unloadCondition
+    this.loaded = false
+  }
+}
+
+export class Asset {
```

Comment:
do we need to declare these as classes? Can we use a functional approach instead?